wp-content/civi-extensions/goonjcustom/Civi/MaterialContributionService.php

```diff
@@ -119,6 +98,63 @@ public static function attachContributionReceiptToEmail(&$params, $context = NUL
     $params['cc'] = 'crm@goonj.org';
   }
 
+  private static function getCityFromGoonjOfficeId($goonjOfficeId) {
+    $city = '';
+    if ($goonjOfficeId) {
+        $organization = Organization::get(FALSE)
+            ->addSelect('address_primary.city')
+            ->addWhere('id', '=', $goonjOfficeId)
+            ->execute()->single();
+
+        if ($organization && isset($organization['address_primary.city'])) {
+            $city = $organization['address_primary.city'];
+        }
+    }
+    return $city;
+  }
+
+  private static function getContributionCity($contribution) {
+    $officeId = $contribution['Material_Contribution.Goonj_Office'];
+
+    if (!$officeId) {
+        // Check for collection camp
+        $activityData = civicrm_api4('Activity', 'get', [
+            'select' => [
+                'Material_Contribution.Collection_Camp',
+            ],
+            'where' => [
+                ['id', '=', $contribution['id']],
+            ],
+            'limit' => 1,
+            'checkPermissions' => FALSE,
+        ]);
```

Comment:
You are using the traditional style of civicrm_api4. Please use OOP style to keep it consistent with the reset of the codebase
<img width="176" alt="image" src="https://github.com/user-attachments/assets/1e6b9851-65d3-415a-a50a-b3e0a6894129">